src/cat.jl

```diff
@@ -28,13 +28,17 @@ end
 
 Base.vcat(a::NamedDimsArray{L}) where {L} = a
 
-for (T, S) in [
-    (:NamedDimsVecOrMat, :NamedDimsVecOrMat),
-    (:NamedDimsVecOrMat, :AbstractVecOrMat),
-    (:AbstractVecOrMat, :NamedDimsVecOrMat),
+for (T, S, U) in [
+    (:NamedDimsVecOrMat, :NamedDimsVecOrMat, :AbstractVecOrMat),
+    (:NamedDimsVecOrMat, :AbstractVecOrMat, :AbstractVecOrMat),
+    (:AbstractVecOrMat, :NamedDimsVecOrMat, :AbstractVecOrMat),
+    # needed since Julia 1.10 because of ambiguities introduced by SparseArrays
+    (:(NamedDimsVecOrMat{<:Any,<:Number}), :(NamedDimsVecOrMat{<:Any,<:Number}), :(AbstractVecOrMat{<:Number})),
+    (:(NamedDimsVecOrMat{<:Any,<:Number}), :(AbstractVecOrMat{<:Number}), :(AbstractVecOrMat{<:Number})),
+    (:(AbstractVecOrMat{<:Number}), :(NamedDimsVecOrMat{<:Any,<:Number}), :(AbstractVecOrMat{<:Number})),
```

Comment:
Sad but probably unavoidable

Reply:
Do you know if the 3rd loop is required to pass?

I would love there to be a better way than overloading `hcat`. At some point wanted to make Base use `axes` everywhere & then perhaps there's a chance you could introduce some pairwise `promote`-like scheme. But... for now we hack things to keep working?

I thought there was talk of removing the offending SparseArrays methods but it looks like nothing has happened... https://github.com/JuliaSparse/SparseArrays.jl/issues/431  is the relevant thread.

Reply:
By 3rd I mean the one over U, instead of `cs::AbstractVecOrMat...` before.

There were `3^2 == 9` methods, PR currently makes `6^3 == 216` instead. (And 4096 in https://github.com/mcabbott/AxisKeys.jl/pull/169 ) I don't know that this causes any problems but it does seem like a lot!

Reply:
Oh I'm sorry, you are right. I thought there were 3 nested loops, but clearly there are not.